### PR TITLE
Jvenstad/badges

### PR DESCRIPTION
**Pull request by jonmv**
@mpolden please review.

### CODE REVIEW

File: controller-api/src/main/java/com/yahoo/vespa/hosted/controller/api/integration/zone/ZoneRegistry.java [hunk 1]
```diff
@@ -75,4 +75,7 @@
     /** Returns a URL used to request support from the Vespa team. */
     URI supportUrl();
 
+    /** Returns a URL used to generate flashy badges from strings. */
+    URI badgeUrl();
```
Comment (mpolden) [inline controller-api/src/main/java/com/yahoo/vespa/hosted/controller/api/integration/zone/ZoneRegistry.java]: Most of these URLs aren't actually zone-related. I think we should pull them out and group them together elsewhere soon (e.g. config).

**Review by mpolden — APPROVED** [review]

File: controller-api/src/main/java/com/yahoo/vespa/hosted/controller/api/integration/zone/ZoneRegistry.java (hunk 1 above)
Comment (jonmv) [reply controller-api/src/main/java/com/yahoo/vespa/hosted/controller/api/integration/zone/ZoneRegistry.java]: Yes.

Comment (jonmv) [reply controller-api/src/main/java/com/yahoo/vespa/hosted/controller/api/integration/zone/ZoneRegistry.java]: They're typically _system_ related. Perhaps we should allow a richer view of a system from the controller (and likewise for a _zone_, but that's just vaguely related)?

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/deployment/JobController.java [hunk 2]
```diff
@@ -314,6 +318,26 @@ public void deactivateTester(TesterId id, JobType type) {
         }
     }
 
+    /** Returns a URI which points at a badge showing historic status of given length for the given job type for the given application. */
+    public URI historicBadge(ApplicationId id, JobType type, int historyLength) {
+        List<Run> runs = new ArrayList<>(runs(id, type).values());
+        Optional<Run> lastCompleted = runs.isEmpty() ? Optional.empty()
```
[the remark this answers is not in the document]
Comment (jonmv) [reply controller-server/src/main/java/com/yahoo/vespa/hosted/controller/deployment/JobController.java]: Ok, can't watch that sad-face any more. Done!

**Pull request merged**